packages/terra-form-checkbox/src/Checkbox.jsx

```diff
@@ -0,0 +1,211 @@
+    */
+  isDisabled: PropTypes.bool,
+  /**
+    * Whether the checkbox element is inline or not.
+    */
+  isInline: PropTypes.bool,
+  /**
+    * Whether the label is hidden or not.
+    */
+  isLabelHidden: PropTypes.bool,
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+  /**
+    * Additional objects for the text object.
+    */
+  labelTextAttrs: PropTypes.object,
+  /**
+    * Name of the input attribute.
+    */
+  name: PropTypes.string,
+  /**
+   * Function to trigger when focus is lost from the checkbox.
+   */
+  onBlur: PropTypes.func,
+  /**
+    * Function to trigger when user clicks on the checkbox. Provide a function to create a controlled input.
+    */
+  onChange: PropTypes.func,
+    /**
+   *  Function to trigger when user focuses on the checkbox.
+   */
+  onFocus: PropTypes.func,
+  /**
+    * The value of the input element.
+    */
+  value: PropTypes.string,
+};
+
+const defaultProps = {
+  checked: undefined,
+  defaultChecked: undefined,
+  id: undefined,
+  inputAttrs: {},
+  isDisabled: false,
+  isInline: false,
+  isLabelHidden: false,
+  labelText: null,
+  labelTextAttrs: {},
+  name: null,
+  onBlur: undefined,
+  onChange: undefined,
+  onFocus: undefined,
+  value: undefined,
+};
+
+class Checkbox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { focus: false };
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+  }
+
+  onFocus() {
+    this.setState({ focus: true });
+    if (this.onFocus !== undefined) {
+      this.onFocus();
+    }
+  }
+
+  onBlur() {
+    this.setState({ focus: false });
+    if (this.onBlur !== undefined) {
+      this.onBlur();
+    }
+  }
+
+  render() {
+    const {
+      checked,
+      defaultChecked,
+      inputAttrs,
+      id,
+      isDisabled,
+      isInline,
+      isLabelHidden,
+      labelText,
+      labelTextAttrs,
+      name,
+      onBlur,
+      onChange,
+      onFocus,
+      value,
+      ...customProps
+    } = this.props;
+
+    const controlInputAttrs = Object.assign({}, inputAttrs);
+
+    if (checked !== undefined) {
+      controlInputAttrs.checked = checked;
+    } else {
+      controlInputAttrs.defaultChecked = defaultChecked;
+    }
+
+    const checkboxClasses = cx([
+      'checkbox',
+      { 'checkbox-inline': isInline },
+      customProps.className,
+    ]);
+
+    const focusContainer = cx([
+      { 'is-hidden-focus-container': isLabelHidden },
+      { 'focus-container': isLabelHidden === false },
+      { focus: this.state.focus },
+    ]);
+
+    const labelClasses = cx([
+      'label',
+      { 'is-disabled': isDisabled },
+      labelTextAttrs.className,
+    ]);
+
+    const inputClasses = cx([
+      'native-input',
+      inputAttrs.className,
+    ]);
+
+    const labelTextClasses = cx([
+      'label-text',
+      { 'is-hidden': isLabelHidden },
+    ]);
+
+    let labelTextContainer = null;
```

Comment:
Updated to do a conditional render based on `isLabelHidden`. I still render a span in both cases, because the css checkbox depends on the span with the `before` and `after` pseudo css selectors. The `labelText`itself changes though - it be set to `aria-label` if `isLabelHidden` is true.